sandboxes/dynamicanalysis/analyze-node.js

```diff
@@ -28,12 +29,61 @@ function install(package) {
   }
 }
 
-function importPkg(package) {
+function importPkg(pkg) {
   try {
-    require(package.name);
+    const mod = require(pkg.name);
+    useModule(mod);
   } catch (e) {
-    console.log(`Failed to import ${package.name}: ${e}`);
-  }  
+    console.log(`Failed to import ${pkg.name}: ${e}`);
+  }
+}
+
+// Adapted from https://stackoverflow.com/a/72326559.
+// Premise is that classes have a non-writable prototype
+function isES6Class(obj) {
+  if (typeof obj !== "function") {
+    return false;
+  }
+
+  const descriptor = Object.getOwnPropertyDescriptor(obj, "prototype");
+
+  return descriptor && !descriptor.writable;
+}
+
+// best-effort execution of code in the module
+function useModule(modulePath) {
+  console.log("[module] " + modulePath);
+
+  //console.log(module);
+
+  // How to tell if something is a function vs class:
+  // - node uses complex internal logic in util.inspect() function [1]
+  // - StackOverflow thread discussing how complex this problem is [2]
+  //
+  // Solution uses code from an answer in [2]
+
+  // [1] https://github.com/nodejs/node/blob/main/lib/internal/util/inspect.js
+  // [2] https://stackoverflow.com/questions/30758961/how-to-check-if-a-variable-is-an-es6-class-declaration/72326559
+  const callableNames = Object.keys(module).filter(
+    (key) => typeof module[key] === "function"
+  );
+
+  // Call all the exported names. If there is a
+  // TypeError, it's probably because it is a class,
+  // so we'll try again using new
```

Comment:
It might be worth calling out some of the limitations with this approach.

e.g. funcs that expect arguments